internal/config/auth_config_test.go

```diff
@@ -103,7 +104,7 @@ func TestTokenFromKeyringNonExistent(t *testing.T) {
 	_, err := authCfg.TokenFromKeyring("github.com")
 
 	// Then it returns failure bubbling the ErrNotFound
-	require.ErrorIs(t, err, keyring.ErrNotFound)
+	require.ErrorContains(t, err, "secret not found in keyring")
```

Comment:
So I'm generally in favour of not tying our `config` to zalando's package, but I do think it would be nice to have a typed error here.

I think especially if we pull in that other keyring PR, having our own keyring abstraction will probably already be valuable.

Reply:
Are you suggesting wrapping the zalando package error in our own error type here? I am good with that, but wonder if we can do that in a follow up PR? I feel like there are a couple different directions we can go with the internal keyring package and don't want to hold up this work based on that.

Reply:
I'm suggesting that our keyring package because a first class abstraction that hides whatever libraries we choose. I'm fine with delaying that. I think that work would go better with the native keyring chunk.